src/lib/controllers/ui-controller.js

```diff
@@ -80,17 +80,27 @@ export default class UIController {
     this.isInitialized = false
     this.isActivated = false
     this.isDeactivated = false
+    this.userDataManager = null
 
     /**
      * A name of the platform (mobile/desktop) UI controller is running within.
      * @type {string} - A platform name from {HTMLPage.platforms}
      */
     this.platform = HTMLPage.getPlatform()
 
+    this.authPlugin = store => {
+      store.subscribe((mutation,state) => {
+        if (mutation.type == 'auth/setIsAuthenticated' || mutation.type == 'auth/setIsNotAuthenticated') {
+            this.userDataManager = this.initUserDataManager()
+        }
+      })
+    }
+
```

Comment:
I'm wondering if the following be simpler:
1. Add `initUserDataManager()` to the `app` public API of a UI controller.
2. Call `initUserDataManager()` from within auth module store mutations?
Then we can avoid using a plugin, and a workflow shall be more straightforward. What do you think?

Reply:
Well, I was wondering about cross-module dependencies. Right now none of the data modules have access to each other.

Reply:
ok, per discussion this is now implemented without a plugin, instead using a watch on the store.auth.isAuthenticated state variable.